implementations/rust/ockam/ockam_api/src/kafka/kafka_portal_listener.rs

```diff
@@ -0,0 +1,85 @@
+use crate::kafka::kafka_inlet_map::KafkaInletMap;
+use crate::kafka::kafka_portal_worker::KafkaPortalWorker;
+use ockam::Context;
+use ockam_core::{Address, AllowAll, Any, Routed, Worker};
+
+
+
+use std::sync::Arc;
+
+use tracing::trace;
+use tracing::warn;
+
+///Acts like an outlet listener of tcp transport
+/// only receives first ping and spawn a dedicated worker to handle the rest
+pub(crate) struct KafkaPortalListener {
+    inlet_map: Arc<KafkaInletMap>,
+}
+
+#[ockam::worker]
+impl Worker for KafkaPortalListener {
+    type Message = Any;
+    type Context = Context;
+
+    async fn handle_message(
+        &mut self,
+        context: &mut Self::Context,
+        message: Routed<Self::Message>,
+    ) -> ockam::Result<()> {
+        trace!("received message: {:?}", &message);
+
+        let result = KafkaPortalWorker::start(context, self.inlet_map.clone()).await;
+        if let Err(err) = result {
+            warn!("cannot spawn worker; err = {:?}", err);
+            return Err(err);
+        }
+        let worker_address = result.unwrap();
```

Comment:
Could you please consider rewriting this part to avoid unwrapping? Since you already handle error condition